### PR TITLE
bump to python3.11

### PR DESCRIPTION
**Pull request by scottyhq**
Updates all images from Python 3.10 -> Python 3.11

Generally I think it's good to keep these images in sync with the latest Python stable version and 3.12 is right around the corner! https://devguide.python.org/versions/

### CODE REVIEW

**Comment by weiji14** [comment]
Couple of things:

- Could also try bumping to Python 3.11 at https://github.com/pangeo-data/pangeo-docker-images/blob/9375b20d1f4c893df43d4703c35029db238e4ee5/environment-condalock.yml#L5
- Running conda-lock on `pangeo-notebook` failed with a super long error message at https://github.com/pangeo-data/pangeo-docker-images/actions/runs/5218659799/jobs/9419649873. The last few lines though are like this:

```
├─ earthdata   is uninstallable because there are no viable options
│  ├─ earthdata 0.1.1a6 would require
│  │  └─ s3fs >=2021.8.1,<2022.0.0  but there are no viable options
│  │     ├─ s3fs [2021.10.0|2021.10.1|2021.8.1|2021.9.0] would require
│  │     │  └─ aiobotocore [1.4.0 |1.4.1 ], which requires
│  │     │     └─ botocore >=1.20.106,<1.20.107 , which conflicts with any installable versions previously reported;
│  │     ├─ s3fs 2021.11.0 would require
│  │     │  ├─ aiobotocore 1.4.1 , which cannot be installed (as previously explained);
│  │     │  └─ fsspec 2021.11.0 , which can be installed;
│  │     └─ s3fs 2021.11.1 would require
│  │        ├─ aiobotocore 2.0.1 , which requires
│  │        │  └─ botocore >=1.22.8,<1.22.9 , which conflicts with any installable versions previously reported;
│  │        └─ fsspec 2021.11.1 , which can be installed;
│  ├─ earthdata [0.2.0|0.2.1|0.2.2|0.3.1] would require
│  │  └─ s3fs >=2021.8.1,<2023.0.0  but there are no viable options
│  │     ├─ s3fs [2021.10.0|2021.10.1|2021.8.1|2021.9.0], which cannot be installed (as previously explained);
│  │     ├─ s3fs 2021.11.0, which cannot be installed (as previously explained);
│  │     ├─ s3fs 2021.11.1, which cannot be installed (as previously explained);
│  │     ├─ s3fs 2022.1.0 would require
│  │     │  ├─ aiobotocore 2.1.0 , which requires
│  │     │  │  └─ botocore >=1.23.24,<1.23.25 , which conflicts with any installable versions previously reported;
│  │     │  └─ fsspec 2022.1.0 , which conflicts with any installable versions previously reported;
│  │     ├─ s3fs 2022.10.0 would require
│  │     │  ├─ aiobotocore >=2.4.0,<2.4.1 , which requires
│  │     │  │  └─ botocore >=1.27.59,<1.27.60 , which conflicts with any installable versions previously reported;
│  │     │  └─ fsspec 2022.10.0 , which conflicts with any installable versions previously reported;
│  │     ├─ s3fs 2022.11.0 would require
│  │     │  ├─ aiobotocore >=2.4.0,<2.4.1 , which cannot be installed (as previously explained);
│  │     │  └─ fsspec 2022.11.0 , which conflicts with any installable versions previously reported;
│  │     ├─ s3fs 2022.2.0 would require
│  │     │  ├─ aiobotocore [2.1.0 |>=2.1.0,<2.2 ], which cannot be installed (as previously explained);
│  │     │  └─ fsspec 2022.2.0 , which conflicts with any installable versions previously reported;
│  │     ├─ s3fs 2022.3.0 would require
│  │     │  ├─ aiobotocore >=2.2.0,<2.3 , which requires
│  │     │  │  └─ botocore >=1.24.21,<1.24.22 , which conflicts with any installable versions previously reported;
│  │     │  └─ fsspec 2022.3.0 , which conflicts with any installable versions previously reported;
│  │     ├─ s3fs 2022.5.0 would require
│  │     │  ├─ aiobotocore >=2.3.0,<2.4 , which cannot be installed (as previously explained);
│  │     │  └─ fsspec 2022.5.0 , which conflicts with any installable versions previously reported;
│  │     ├─ s3fs 2022.7.1 would require
│  │     │  ├─ aiobotocore >=2.3.4,<2.4 , which cannot be installed (as previously explained);
│  │     │  └─ fsspec 2022.7.1 , which conflicts with any installable versions previously reported;
│  │     └─ s3fs 2022.8.2 would require
│  │        ├─ aiobotocore >=2.4.0,<2.4.1 , which cannot be installed (as previously explained);
│  │        └─ fsspec 2022.8.2 , which conflicts with any installable versions previously reported;
│  └─ earthdata [0.4.0|0.4.1] would require
│     ├─ fsspec >=2022.1 , which conflicts with any installable versions previously reported;
│     ├─ python-benedict >=0.25 , which requires
│     │  └─ pyyaml >=6.0,<7.0  with the potential options
│     │     ├─ pyyaml [5.4.1|6.0], which can be installed (as previously explained);
│     │     ├─ pyyaml 6.0 conflicts with any installable versions previously reported;
│     │     ├─ pyyaml [5.3|5.3.1|5.4.1|6.0], which can be installed (as previously explained);
│     │     ├─ pyyaml [5.3.1|5.4.1|6.0], which can be installed (as previously explained);
│     │     ├─ pyyaml [5.3|5.3.1|5.4.1|6.0], which can be installed (as previously explained);
│     │     ├─ pyyaml [5.4.1|6.0], which cannot be installed (as previously explained);
│     │     ├─ pyyaml [5.3.1|5.4.1|6.0], which can be installed (as previously explained);
│     │     └─ pyyaml [5.4.1|6.0], which cannot be installed (as previously explained);
│     └─ s3fs >=2021.11,<2024  but there are no viable options
│        ├─ s3fs 2021.11.0, which cannot be installed (as previously explained);
│        ├─ s3fs 2021.11.1, which cannot be installed (as previously explained);
│        ├─ s3fs 2022.1.0, which cannot be installed (as previously explained);
│        ├─ s3fs 2022.10.0, which cannot be installed (as previously explained);
│        ├─ s3fs 2022.11.0, which cannot be installed (as previously explained);
│        ├─ s3fs 2022.2.0, which cannot be installed (as previously explained);
│        ├─ s3fs 2022.3.0, which cannot be installed (as previously explained);
│        ├─ s3fs 2022.5.0, which cannot be installed (as previously explained);
│        ├─ s3fs 2022.7.1, which cannot be installed (as previously explained);
│        ├─ s3fs 2022.8.2, which cannot be installed (as previously explained);
│        ├─ s3fs 2023.1.0 would require
│        │  ├─ aiobotocore >=2.4.2,<2.4.3 , which cannot be installed (as previously explained);
│        │  └─ fsspec 2023.1.0 , which conflicts with any installable versions previously reported;
│        ├─ s3fs 2023.3.0 would require
│        │  ├─ aiobotocore >=2.4.2,<2.4.3 , which cannot be installed (as previously explained);
│        │  └─ fsspec 2023.3.0 , which conflicts with any installable versions previously reported;
│        └─ s3fs [2023.4.0|2023.5.0] would require
│           └─ aiobotocore >=2.5.0,<2.5.1 , which requires
│              └─ botocore >=1.29.76,<1.29.77 , which conflicts with any installable versions previously reported;
├─ pangeo-notebook 2023.05.30**  is installable and it requires
│  └─ pangeo-dask 2023.05.30.* , which requires
│     └─ dask 2023.5.1.* , which requires
│        └─ lz4 >=4.3.2  with the potential options
│           ├─ lz4 4.3.2 would require
│           │  └─ python_abi 3.10.* *_cp310 with the potential options
│           │     ├─ python_abi 3.10, which can be installed;
│           │     └─ python_abi 3.10, which cannot be installed (as previously explained);
│           ├─ lz4 4.3.2 would require
│           │  └─ python_abi 3.11.* *_cp311, which can be installed;
│           ├─ lz4 4.3.2 would require
│           │  └─ python_abi 3.8.* *_cp38 with the potential options
│           │     ├─ python_abi 3.8, which can be installed;
│           │     └─ python_abi 3.8, which cannot be installed (as previously explained);
│           ├─ lz4 4.3.2 would require
│           │  └─ python_abi 3.8 *_pypy38_pp73, which cannot be installed (as previously explained);
│           ├─ lz4 4.3.2 would require
│           │  └─ python_abi 3.9.* *_cp39 with the potential options
│           │     ├─ python_abi 3.9, which can be installed;
│           │     └─ python_abi 3.9, which cannot be installed (as previously explained);
│           └─ lz4 4.3.2 would require
│              └─ python_abi 3.9 *_pypy39_pp73, which cannot be installed (as previously explained);
├─ python 3.11** , which cannot be installed (as previously explained);
└─ xcape   is installable with the potential options
   ├─ xcape [0.1|0.1.1|0.1.2|0.1.3|0.1.4] would require
   │  └─ python_abi 3.6.* *_cp36m, which can be installed;
   ├─ xcape [0.1|0.1.1|0.1.2|0.1.3|0.1.4] would require
   │  └─ python_abi 3.7.* *_cp37m with the potential options
   │     ├─ python_abi 3.7, which can be installed;
   │     └─ python_abi 3.7, which cannot be installed (as previously explained);
   ├─ xcape [0.1.1|0.1.2|0.1.3|0.1.4] would require
   │  └─ python_abi 3.8.* *_cp38 with the potential options
   │     ├─ python_abi 3.8, which can be installed;
   │     └─ python_abi 3.8, which cannot be installed (as previously explained);
   ├─ xcape [0.1.2|0.1.3|0.1.4] would require
   │  └─ python_abi 3.9.* *_cp39 with the potential options
   │     ├─ python_abi 3.9, which can be installed;
   │     └─ python_abi 3.9, which cannot be installed (as previously explained);
   ├─ xcape 0.1.4 would require
   │  └─ python_abi 3.10.* *_cp310 with the potential options
   │     ├─ python_abi 3.10, which can be installed;
   │     └─ python_abi 3.10, which cannot be installed (as previously explained);
   ├─ xcape 0.1.4 would require
   │  └─ python_abi 3.11.* *_cp311, which can be installed;
   ├─ xcape 0.1.4 would require
   │  └─ python_abi 3.8 *_pypy38_pp73, which cannot be installed (as previously explained);
   └─ xcape 0.1.4 would require
      └─ python_abi 3.9 *_pypy39_pp73, which cannot be installed (as previously explained).
```

Thinking that renaming `earthdata` to `earthaccess` might help a bit, see PR at #463.

**Comment by scottyhq** [comment]
For `forge` image looks like apache-beam-with-gcp>=2.47 is needed 

[apache-beam-with-gcp=2.43.0](https://github.com/pangeo-data/pangeo-docker-images/blob/9375b20d1f4c893df43d4703c35029db238e4ee5/forge/environment.yml#LL12C4-L12C31)

```
apache-beam-with-gcp          2.45.0  py39hf3d152e_0  conda-forge 
apache-beam-with-gcp          2.46.0 py310hff52083_0  conda-forge         
apache-beam-with-gcp          2.46.0  py38h578d9bd_0  conda-forge         
apache-beam-with-gcp          2.46.0  py39hf3d152e_0  conda-forge         
apache-beam-with-gcp          2.47.0 py310hff52083_0  conda-forge         
apache-beam-with-gcp          2.47.0 py311h38be061_0  conda-forge         
apache-beam-with-gcp          2.47.0  py38h578d9bd_0  conda-forge         
apache-beam-with-gcp          2.47.0  py39hf3d152e_0  conda-forge         
apache-beam-with-gcp          2.48.0 py310hff52083_0  conda-forge         
apache-beam-with-gcp          2.48.0 py311h38be061_0  conda-forge 
```

**Comment by scottyhq** [comment]
For `ml-notebook` waiting on `tensorflow>=2.12` which supports python 3.11 https://github.com/conda-forge/tensorflow-feedstock/pull/316

**Comment by weiji14** [comment]
> For `forge` image looks like apache-beam-with-gcp>=2.47 is needed

Started PR to pin to `apache-beam-with-gcp` at #470

> For `ml-notebook` waiting on `tensorflow>=2.12` which supports python 3.11 [conda-forge/tensorflow-feedstock#316](https://github.com/conda-forge/tensorflow-feedstock/pull/316)

New effort going on at https://github.com/conda-forge/tensorflow-feedstock/pull/323.

File: pangeo-notebook/packages.txt
Comment (scottyhq) [inline pangeo-notebook/packages.txt:134]: significant drop here. i'm not familiar with this library...

File: pangeo-notebook/packages.txt
Comment (scottyhq) [inline pangeo-notebook/packages.txt:428]: not sure why pandas would go from 2->1

Comment (weiji14) [reply pangeo-notebook/packages.txt:428]: erddapy-0.7.2 uses `pandas >=0.20.3,<2a0` :upside_down_face:

File: pangeo-notebook/packages.txt
Comment (weiji14) [reply pangeo-notebook/packages.txt:134]: I managed to force `erddapy>=2.2.0` in the environment.yml file and lock it with `conda-lock=2.3` locally. Let's try running `/condalock` again?

Comment (weiji14) [reply pangeo-notebook/packages.txt:134]: Yeah, still not working. Maybe it's because we're still using `conda-lock=1.4` here?

https://github.com/pangeo-data/pangeo-docker-images/blob/48c911ca7f917d9b480fa9f2cfd8e4bf0c45e75d/environment-condalock.yml#L5-L6

Let me open another PR to bump up the conda-lock version.

Comment (weiji14) [reply pangeo-notebook/packages.txt:134]: Hmm, erddapy is still locked to 0.7.2. Maybe we should manually set a minimum pin to `erddapy>=2.2.20` here

https://github.com/pangeo-data/pangeo-docker-images/blob/f6c380c948de63fe4bec01895062291cf7dcacb3/pangeo-notebook/environment.yml#L21

Comment (scottyhq) [reply pangeo-notebook/packages.txt:134]: I think argopy is actually the issue: 

`conda-tree -n testpy11 whoneeds erddapy -t --small` 
```
erddapy==0.7.2
  └─ argopy 0.1.6 [required: >=0.5,<0.8.0]
```

`conda search argopy --info`
NOTE: More recent versions of argopy pin `- gsw <=3.4.0` 

But we don't have `gsw` pinned and this is a decision point for conda-lock: Either 1) use an new version of `gsw` and pin argopy/erddapy to older versions. or (2) pin old gsw and use newer erddapy/argopy.

Comment (scottyhq) [reply pangeo-notebook/packages.txt:134]: opened https://github.com/conda-forge/argopy-feedstock/issues/15

Comment (ocefpaf) [reply pangeo-notebook/packages.txt:134]: > NOTE: More recent versions of argopy pin `- gsw <=3.4.0`

I don't think this pin is needed in argopy b/c I don't believe we had any changes in gsw that could change how argopy works from 3.4 to 3.6 we have mostly bugfixes. Also,it is highly recommended to use gsw >=3.6 due to these bugfixes in the underlying C-library.

Comment (scottyhq) [reply pangeo-notebook/packages.txt:134]: until the feedstock is fixed we can get argopy from pip

File: pangeo-notebook/environment.yml
Comment (scottyhq) [inline pangeo-notebook/environment.yml:1]: In the interest of keeping things moving forward removed version pins, packages that have been commented out for a while (and no complaints so I assume folks aren't using those libraries), and removed `jupyterlab-s3-browser` as it seems unmaintained (https://github.com/IBM/jupyterlab-s3-browser). Getting argopy from pip until conda-forge feedstock is fixed. Tried to keep alphabetized.

File: pangeo-notebook/packages.txt
Comment (scottyhq) [inline pangeo-notebook/packages.txt:480]: Now part of shapely 2.0 (https://github.com/pygeos/pygeos)

File: pangeo-notebook/environment.yml
Comment (weiji14) [reply pangeo-notebook/environment.yml:1]: Sounds good, but can you add back `jupyterlab-git`?

> Removed `jupyterlab-s3-browser` as it seems unmaintained (https://github.com/IBM/jupyterlab-s3-browser).

Mentioned at https://github.com/pangeo-data/pangeo-docker-images/issues/383 that maybe https://github.com/Navteca/jupyterlab-bxplorer is more maintained? But let's add it separately.

**Review by weiji14 — APPROVED** [review]
Wish the diff could be smaller, but looks good to me! Thanks @scottyhq for seeing this through. Will leave it up to you on whether to merge this before or after https://github.com/conda-forge/argopy-feedstock/issues/15 is done.

**Pull request merged**
